tests/queries/0_stateless/02346_text_index_hint_map.sh

```diff
@@ -15,8 +15,8 @@ $MY_CLICKHOUSE_CLIENT --query "
     CREATE TABLE tab
     (
         m Map(String, String),
-        INDEX idx_mk (mapKeys(m)) TYPE text(tokenizer = array) GRANULARITY 4,
-        INDEX idx_mv (mapValues(m)) TYPE text(tokenizer = array) GRANULARITY 4
+        INDEX idx_mk (mapKeys(m)) TYPE text(tokenizer = sparseGrams) GRANULARITY 4,
```

Comment:
`array` changed to `sparseGrams` because the former doesn't support LIKE matching, right?

We should actually use the simplest tokenizer which supports LIKE and that is `splitByNonAlpha`.